src/parser/jobs/mnk/modules/PerfectBalance.tsx

```diff
@@ -27,48 +36,164 @@ const PB_BAD_ACTIONS: ActionKey[] = [
 
 interface Balance {
 	bads: number
-	stacks: number
+	stacks: number // this is the number of stacks the window began with (may be less than the max for dungeons/24mans with a carried-over PB)
 	used: number
+	start: number
 }
 
-export class PerfectBalance extends Analyser {
+const OPO_GAUGE_COLOR = Color('#a256dc')
+const RAPTOR_GAUGE_COLOR = Color('#57b39a')
+const COEURL_GAUGE_COLOR = Color('#d7548e')
+const LUNAR_NADI_COLOR = Color('#8a57c4')
+const SOLAR_NADI_COLOR = Color('#e7dbd3')
+
+export class PerfectBalance extends Gauge {
 	static override debug = false
 	static override handle = 'perfectBalance'
 	static override title = t('mnk.pb.title')`Perfect Balance`
 	static override displayOrder = DISPLAY_ORDER.PERFECT_BALANCE
 
-	@dependency private data!: Data
 	@dependency private suggestions!: Suggestions
 
 	private badActions: Array<Action['id']> = []
 	private formActions: Array<Action['id']> = []
+	private opoActions: Array<Action['id']> = []
+	private raptorActions: Array<Action['id']> = []
+	private coeurlActions: Array<Action['id']> = []
+	private blitzActions: Array<Action['id']> = []
 
 	private current: Balance | undefined
 	private history: Balance[] = []
 
 	private perfectHook?: EventHook<Events['action']>
+	private beastTimeoutHook?: TimestampHook
+
+	private opoBeastGauge = this.add(new CounterGauge({
+		maximum: 3,
+		initialValue: 0,
+		graph: {
+			handle: BEAST_GAUGE_HANDLE,
+			label: <Trans id="mnk.gauge.resource.beast.opo">Opo-opo</Trans>,
+			color: OPO_GAUGE_COLOR.fade(GAUGE_FADE),
+		},
+	}))
```

Comment:
Not sure exactly where or why this happens but the Beast Gauge resource group seems to default to "Opo-opo" as its title (because it's the first gauge added maybe?):

![image](https://github.com/xivanalysis/xivanalysis/assets/65056303/122436e4-1a57-430a-9374-bcbfc21cc323)

Ideally this would be called "Beast Chakra" or something instead.

Reply:
Hm, it should have been labeled that way, I probably goofed something in the un-stashing/detangling from the Enum gauge PR (which definitely will fix this) but I'll get that corrected in case this one gets merged first

Reply:
Yep, lost the resource group definition somewhere along the way. Fixed now.